scapy/layers/l2.py

```diff
@@ -539,11 +539,12 @@ class EAP_MD5(EAP):
         ByteEnumField("code", 1, eap_codes),
         ByteField("id", 0),
         FieldLenField("len", None, fmt="H", length_of="optional_name",
-                      adjust=lambda p, x: x + p.value_size + 6),
+                adjust=lambda p, x: (x + p.value_size + 6) if p.value_size is not None else 6),
```

Comment:
Thanks for the review !

I have fixed the lambda in the `len` field of the `EAP_MD5` class, and rewritten the lambdas in the `EAP_TLS` and `EAP_FAST` as well.